x-pack/plugins/fleet/common/services/policy_template.ts

```diff
@@ -22,7 +22,7 @@ const DATA_STREAM_DATASET_VAR: RegistryVarsEntry = {
   type: 'text',
   title: 'Dataset name',
   description:
-    "Set the name for your dataset. Once selected a dataset cannot be changed without creating a new integration policy. You can't use `-` in the name of a dataset and only valid characters for [Elasticsearch index names](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html) are permitted.\n",
+    "Set the name for your dataset. Once selected a dataset cannot be changed without creating a new integration policy. You can't use `-` in the name of a dataset and only valid characters for [Elasticsearch index names](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html) are permitted.\n",
```

Comment:
updated this link with the correct page that has elasticsearch index validation rules
for some reason I don't see it updated on the UI
<img width="982" alt="image" src="https://github.com/elastic/kibana/assets/90178898/96cdcc03-51c8-41e1-9ca2-5b9a8b2b8e6f">